components/script/script_thread.rs

```diff
@@ -869,6 +867,52 @@ impl ScriptThread {
         });
     }
 
+    /// Step 13 of https://html.spec.whatwg.org/multipage/#navigate
+    pub fn navigate(parent_pipeline_id: PipelineId, mut load_data: LoadData, replace: bool) {
+        SCRIPT_THREAD_ROOT.with(|root| {
```

Comment:
Rename `parent_pipeline_id` (which is normally used to mean the pipeline id of an iframe's parent) to `pipeline_id`.